### PR TITLE
Issue #13100: Create tests from ConstantName xdoc examples

### PR DESCRIPTION
**Pull request by stoyanK7**
Part of #13100

### CODE REVIEW

**Comment by stoyanK7** [comment]
Github, generate site

**Review by romani — APPROVED** [review]
ok to merge where CI is green

File: src/xdocs-test/resources/com/puppycrawl/tools/checkstyle/checks/naming/constantname/Example1.java
Comment (nrmancuso) [inline src/xdocs-test/resources/com/puppycrawl/tools/checkstyle/checks/naming/constantname/Example1.java:1]: `xdocs-test` -> `xdocs-examples`, per https://github.com/checkstyle/checkstyle/pull/13112#discussion_r1215588062

**Review by nrmancuso — CHANGES_REQUESTED** [review]
Item:

File: src/xdocs-test/resources/com/puppycrawl/tools/checkstyle/checks/naming/constantname/Example1.java
Comment (stoyanK7) [reply src/xdocs-test/resources/com/puppycrawl/tools/checkstyle/checks/naming/constantname/Example1.java:1]: @nrmancuso Done

**Review by nrmancuso — APPROVED** [review]

**Comment by stoyanK7** [comment]
Moving to draft until https://github.com/checkstyle/checkstyle/pull/13187 gets merged. Then I will rebase and move to ready for review again so it doesn't get merged by mistake.

**Comment by romani** [comment]
Rebased by GitHub

**Comment by romani** [comment]
GitHub, generate website

File: src/xdocs-examples/java/com/puppycrawl/tools/checkstyle/checks/naming/ConstantNameCheckExamplesTest.java [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+            "21:20: " + getCheckMessage(MSG_INVALID_PATTERN, "loggerMYSELF", DEFAULT_PATTERN),
+            "23:30: " + getCheckMessage(MSG_INVALID_PATTERN, "myselfConstant", DEFAULT_PATTERN),
+        };
+
+        verifyWithInlineConfigParser(getPath("Example1.java"), expected);
+    }
+
+    @Test
+    public void testExample2() throws Exception {
+        final String format = "^log(ger)?$|^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$";
```
Comment (nrmancuso) [inline src/xdocs-examples/java/com/puppycrawl/tools/checkstyle/checks/naming/ConstantNameCheckExamplesTest.java:57]: `format` -> `pattern`, if we have `DEFAULT_PATTERN` it makes sense to stick with *pattern naming

**Review by nrmancuso — CHANGES_REQUESTED** [review]
Minor item:

File: src/xdocs-examples/java/com/puppycrawl/tools/checkstyle/checks/naming/ConstantNameCheckExamplesTest.java (hunk 1 above)
Comment (stoyanK7) [reply src/xdocs-examples/java/com/puppycrawl/tools/checkstyle/checks/naming/ConstantNameCheckExamplesTest.java]: Done. I corrected the same mistake in `AbstractClassName` PR as well.

**Review by nrmancuso — APPROVED** [review]

**Review by romani — APPROVED** [review]

**Comment by rnveach** [comment]
I still see updating the xdoc as pointless if we are just going to remove them.

I will not be validating the conversion process in these PRs (java and config were transferred correctly and it is still the same example). I am expecting @romani and/or @nrmancuso to do that as it is too manual.

**Review by rnveach — APPROVED** [review]

**Comment by romani** [comment]
>I still see updating the xdoc as pointless if we are just going to remove them.

We will remove them only after we finish bunch of automation, we are not yet close to this time.

**Comment by romani** [comment]
@stoyanK7 , looks like we all aligned on final view.
Please proceed with conversion of xdoc java snippets to Examples.  I am not sure what is best approach: manual or semi-automated.
Please share your vision on this, in separate issue.
Better to not continue discussion in this PR.

**Pull request merged**